### ISSUE TITLE
[Bug]: Error when creating a topic with an underscore and/or an uppercase in the name

### ISSUE THREAD

**Issue by morenol** [infinyon/fluvio#1464]
When I try to create a topic with an uppercase letter in the name I got this error:

```bash
Error: 
   0: Consumer Error
   1: Aug 19 15:26:09.586 DEBUG fluvio_socket::multiplexing: multiplexer terminated
Fluvio client error
   2: Fluvio socket error
   3: time out in serial: 1001 request: 4Aug 19 15:26:09.586 DEBUG fluvio::sync::controller: Topic received request
```

I see this in the SC logs
```
Aug 19 20:22:30.579 ERROR K8StateDispatcher{spec="Topic" namespace="acct-76ddb9e8-257e-4083-ae3b-8259065655e3"}:process_ws_action: fluvio_stream_dispatcher::dispatcher::k8_dispatcher: error: Topic, update spec client error: 422 Unprocessable Entity
```

K8 related documentation https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

**Comment by sehz** [comment]
- [ ] In the SC, reject any name that doesn't conform to current naming convention
- [ ] Client API, returns appropriate Error
- [ ] CLI, show right error message

**Comment by morenol** [comment]
Closed by #1587
